internal/pkg/config/app.go

```diff
@@ -14,11 +14,12 @@ import (
 
 // Application is a named collection of environments and services.
 type Application struct {
-	Name      string            `json:"name"`           // Name of an Application. Must be unique amongst other apps in the same account.
-	AccountID string            `json:"account"`        // AccountID this app is mastered in.
-	Domain    string            `json:"domain"`         // Existing domain name in Route53. An empty domain name means the user does not have one.
-	Version   string            `json:"version"`        // The version of the app layout in the underlying datastore (e.g. SSM).
-	Tags      map[string]string `json:"tags,omitempty"` // Labels to apply to resources created within the app.
+	Name               string            `json:"name"`               // Name of an Application. Must be unique amongst other apps in the same account.
+	AccountID          string            `json:"account"`            // AccountID this app is mastered in.
+	Domain             string            `json:"domain"`             // Existing domain name in Route53. An empty domain name means the user does not have one.
+	DomainHostedZoneID string            `json:"domainHostedZoneID"` // Existing domain hosted zone in Route53. An empty domain name means the user does not have one.
```

Comment:
I am slightly worried that for existing customers this will be a lot of work to retrofit on their own.

Do you think we can do something similar to `env upgrade` but `app upgrade` instead? that can add this field and also update the stackset instaces with the new permission?

Not in this PR, but I think it's worthwhile doing a short design doc on this.
```
`app upgrade` is run both within the buildspec and `copilot deploy`
   If there is no new Version of the app, then do nothing.
   Otherwise update schema, and update stackset instances.
``` 
(something like this)

Reply:
Discussed offline. Will be working on `app upgrade` after alias implementation but before making it visible to customers.